adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala

```diff
@@ -122,16 +135,16 @@ object MdTag {
                     mdTagStringOffset += deletedBases.size
                   }
                 }
-                cigarReferencePosition += cigarElement.getLength
                 cigarIdx += 1
+                cigarOperatorIndex = 0
               }
               case _ => throw new IllegalArgumentException(s"CIGAR ${cigar.toString} indicates deletion found but no deleted bases in MDTag $mdTagInput")
             }
           }
           case _ if cigarElement.getOperator.consumesReferenceBases() => {
             referencePos += cigarElement.getLength
-            cigarReferencePosition += cigarElement.getLength
             cigarIdx += 1
+            cigarOperatorIndex = 0
           }
           case _ => {
             cigarIdx += 1
```

Comment:
`cigarOperatorIndex = 0` is not necessary, it shouldn't be updated, but just added it for clarity.

The last case is insertion or clipped bases